code/datums/bodypart_overlays/bodypart_overlay.dm

```diff
@@ -0,0 +1,82 @@
+///Bodypart ovarlay datum. These can be added to any limb to give them a proper overlay, that'll even stay if the limb gets removed
+///This is the abstract parent, don't use it!!
+/datum/bodypart_overlay
+	///Sometimes we need multiple layers, for like the back, middle and front of the person (EXTERNAL_FRONT, EXTERNAL_ADJACENT, EXTERNAL_BEHIND)
+	var/layers
+	///List of all possible layers. Used for looping through in drawing
+	var/static/list/all_layers = list(EXTERNAL_FRONT, EXTERNAL_ADJACENT, EXTERNAL_BEHIND, EXTERNAL_FRONT_UNDER_CLOTHES, EXTERNAL_FRONT_OVER) // SKYRAT EDIT - Customization - ORIGINAL: var/static/list/all_layers = list(EXTERNAL_FRONT, EXTERNAL_ADJACENT, EXTERNAL_BEHIND)
+
+	///Key of the icon states of all the sprite_datums for easy caching
+	var/cache_key = ""
+
+///Wrapper for getting the proper image, colored and everything
+/datum/bodypart_overlay/proc/get_overlay(layer, obj/item/bodypart/limb)
+	layer = bitflag_to_layer(layer)
+	. = get_image(layer, limb)
+	color_image(., layer, limb)
+
+///Generate the image. Needs to be overriden
+/datum/bodypart_overlay/proc/get_image(layer, obj/item/bodypart/limb)
+	CRASH("Get image needs to be overridden")
+
+///Color the image
+/datum/bodypart_overlay/proc/color_image(image/overlay, layer)
+	return
+
+///Called on being added to a limb
+/datum/bodypart_overlay/proc/added_to_limb(obj/item/bodypart/limb)
+	return
+
+///Called on being removed from a limb
+/datum/bodypart_overlay/proc/removed_from_limb(obj/item/bodypart/limb)
+	return
+
+///Use this to change the appearance (and yes you must overwrite hahahahahah) (or dont use this, I just dont want people directly changing the image)
+/datum/bodypart_overlay/proc/set_appearance()
+	CRASH("Update appearance needs to be overridden")
+
+/**This exists so sprite accessories can still be per-layer without having to include that layer's
+*  number in their sprite name, which causes issues when those numbers change.
+*/
+/datum/bodypart_overlay/proc/mutant_bodyparts_layertext(layer)
+	switch(layer)
+		if(BODY_BEHIND_LAYER)
+			return "BEHIND"
+		if(BODY_ADJ_LAYER)
+			return "ADJ"
+		if(BODY_FRONT_LAYER)
+			return "FRONT"
+		//SKYRAT EDIT ADDITION - Customization
+		if(BODY_FRONT_UNDER_CLOTHES)
+			return "FRONT_UNDER"
+		if(ABOVE_BODY_FRONT_HEAD_LAYER)
+			return "FRONT_OVER"
+		//SKYRAT EDIT ADDITION END
+
+///Converts a bitflag to the right layer. I'd love to make this a static index list, but byond made an attempt on my life when i did
+/datum/bodypart_overlay/proc/bitflag_to_layer(layer)
+	switch(layer)
+		if(EXTERNAL_BEHIND)
+			return BODY_BEHIND_LAYER
+		if(EXTERNAL_ADJACENT)
+			return BODY_ADJ_LAYER
+		if(EXTERNAL_FRONT)
+			return BODY_FRONT_LAYER
+		//SKYRAT EDIT ADDITION - Customization
```

Comment:
ditto...?

Reply:
This was in the missed mirror, and I then added some changes myself.